server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -293,7 +299,8 @@ public void updateState(ClusterState state) {
             bestGenerationFromCS = bestGeneration(cleanupInProgress.entries());
         }
 
-        final long finalBestGen = bestGenerationFromCS;
+        metadata = getRepoMetaData(state);
+        final long finalBestGen = Math.max(bestGenerationFromCS, metadata.generation());
```

Comment:
I figured it was nice to still use them as a fall-back for the BwC case. Though ... come to think of it, they might not matter since we don't actually have any logic where a non-master node reads the `RepositoryData` as far as I can tell. So it really only helps in the corner case of failing over from an older version master node => I guess I don't see the harm in keeping it around for `7.6`, but I'm fine dropping it as well :)